bot.py

```diff
@@ -140,63 +160,90 @@ def pr_url(self):
             "And by 'released', I mean completely...um...leased.".format(self.version)
         )
 
-    def message_if_unchecked(self):
+    async def message_if_unchecked(self):
         """
         Send a message next morning if any boxes are not yet checked off
         """
         unchecked_authors = get_unchecked_authors(self.org, self.repo, self.version)
         if unchecked_authors:
             slack_usernames = self.translate_slack_usernames(unchecked_authors)
-            self.say(
+            await self.say(
                 "What an unexpected surprise! "
                 "The following authors have not yet checked off their boxes: {}".format(
                     ", ".join(slack_usernames)
                 )
             )
 
+    async def delay_message(self):
+        now = datetime.now()
+        tomorrow_at_10 = next_workday_at_10(now)
+        await asyncio.sleep((tomorrow_at_10 - now).total_seconds())
+        await self.message_if_unchecked()
+
+
+def get_version_number(message):
+    return re.findall('[0-9.]+', message)
+
+mm_urls = {
+    'repo_dir': os.environ.get('REPO_DIR'),
+    'rc_hash_url': "https://micromasters-rc.herokuapp.com/static/hash.txt",
+    'prod_hash_url': "https://micromasters.mit.edu/static/hash.txt"
+}
+channel_mapping = {
+    'C67SGCU9H': mm_urls
+}
+
 
 def main():
     """main function for bot command"""
-    parser = argparse.ArgumentParser()
-    parser.add_argument("command", choices=['release', 'wait_for_checkboxes', 'finish_release'])
-    parser.add_argument("repo_dir")
-    parser.add_argument("version")
-    parser.add_argument("--org", default="mitodl")
-    parser.add_argument("--rc-hash-url", default="https://micromasters-rc.herokuapp.com/static/hash.txt")
-    parser.add_argument("--prod-hash-url", default="https://micromasters.mit.edu/static/hash.txt")
-    args = parser.parse_args()
-
-    slack_webhook_url = os.environ.get('SLACK_WEBHOOK_URL')
-    if not slack_webhook_url:
-        raise Exception("Missing SLACK_WEBHOOK_URL")
-
     slack_access_token = os.environ.get('SLACK_ACCESS_TOKEN')
     if not slack_access_token:
         raise Exception("Missing SLACK_ACCESS_TOKEN")
 
-    bot = Bot(slack_webhook_url, slack_access_token, args.repo_dir, args.version, args.rc_hash_url, args.prod_hash_url)
-
-    now = datetime.now()
-    tomorrow_at_10 = next_workday_at_10(now)
+    bot_access_token = os.environ.get('BOT_ACCESS_TOKEN')
+    mm_channel = 'C67SGCU9H'
+
+    resp = requests.post("https://slack.com/api/rtm.connect", data={
+        "token": bot_access_token,
+    })
+
+    async def connect_to_message_server(loop):
+        async with websockets.connect(resp.json()['url']) as websocket:
+            bot = Bot(websocket, slack_access_token, mm_channel)
+
+            while True:
+                message = await websocket.recv()
+                print(message)
+                message = json.loads(message)
+
+                if message.get('type') == 'desktop_notification':
```

Comment:
I don't think we shouldn't check `desktop_notification`, we should check the `message` message instead. I'm not seeing `desktop_notification` in the list in [their docs](https://api.slack.com/events) so I'm not sure if the `desktop_notification` event is meant to be relied on or not. The `text` part of the message is `<@U68PT1K47> hi` where `<@U68PT1K47>` is the marked up equivalent to `@doof`.

Can you change this to:
 - Look for messages starting with the user tag of the bot. You can get the user id from the `rtm.connect` response
 - Add a function `handle_message` to `Bot` which takes the rest of the message after the `<@U...> ` part, and move the rest of this code there